### ISSUE TITLE
Missing init on destructuring not flagged as syntax error when inside for loop

### ISSUE THREAD

**Issue by marvinhagemeister** [oxc-project/oxc#8220]
This code should be a syntax error, but is parsed successfully in OXC. Other parsers like Babel and SWC throw a syntax error as expected.

```ts
for (let { x };;);
```

Interestingly, OXC seems to be able to detect the missing initializer properly when it's outside of the for loop:

```ts
// throws: Error: Missing initializer in destructuring declaration
let { x }
```

Babel:

```sh
Missing initializer in destructuring declaration. (1:14)

> 1 | for (let { x };;);
    |               ^
```

SWC:

```sh
  x Destructuring bindings require initializers
   ,-[input.tsx:1:1]
 1 | for (let { x };;);
   :          ^^^^^
   `----
```
